engine/src/flutter/lib/ui/painting/display_list_deferred_image_gpu_impeller.h

```diff
@@ -86,27 +93,24 @@ class DlDeferredImageGPUImpeller final : public DlImage {
     std::optional<std::string> get_error();
 
    private:
+    FML_FRIEND_TEST(testing::DlDeferredImageGPUImpeller, TrashesDisplayList);
     DlISize size_;
-    sk_sp<DisplayList> display_list_;
     std::shared_ptr<impeller::Texture> texture_;
     fml::TaskRunnerAffineWeakPtr<SnapshotDelegate> snapshot_delegate_;
     fml::RefPtr<fml::TaskRunner> raster_task_runner_;
-    std::shared_ptr<TextureRegistry> texture_registry_;
 
     mutable std::mutex error_mutex_;
     std::optional<std::string> error_;
 
     ImageWrapper(
-        sk_sp<DisplayList> display_list,
         const DlISize& size,
         fml::TaskRunnerAffineWeakPtr<SnapshotDelegate> snapshot_delegate,
         fml::RefPtr<fml::TaskRunner> raster_task_runner);
 
-    // If a layer tree is provided, it will be flattened during the raster
-    // thread task spawned by this method. After being flattened into a display
-    // list, the image wrapper will be updated to hold this display list and the
-    // layer tree can be dropped.
-    void SnapshotDisplayList(std::unique_ptr<LayerTree> layer_tree = nullptr);
+    // If a layer tree is provided, it will be flattened into a display list
+    // during the raster thread task spawned by this method.
+    void SnapshotDisplayList(
+        std::variant<sk_sp<DisplayList>, std::unique_ptr<LayerTree>> content);
```

Comment:
Looking at this now, we probably should have just overloaded the constructor. Oh well, that was fun.